jython/DispatcherSystem/CreateIcons.py

```diff
@@ -443,13 +443,28 @@ def removeTransits(self):
     # remove Sections
     # **************************************************
     def removeSections(self):
+
+        #remove sections
         deleteList = []     # Prevent concurrent modification
         for section in sections.getNamedBeanSet():
             deleteList.append(section)
```

Comment:
An alternate approach is to have a `directionSensorDeleteList` and add the direction sensors, if any, to the list as you process each section. 

sensor = section.getForwardBlockingSensor()
if forward is not None:
    directionSensorDeleteList.add(forward)

Repeat for getReverseBlockingSensor.


If the direction sensors have been added to the SML, then the sensor deletes have to wait until after the SML deletes.